pendulum_control/src/pendulum_demo.cpp

```diff
@@ -293,12 +251,25 @@ int main(int argc, char * argv[])
       }
     };
 
+  // Create a lambda function that will fire regularly to publish the next results message.
+  auto results_msg = std::make_shared<pendulum_msgs::msg::RttestResults>();
+  auto logger_publish_callback =
+    [&logger_pub, &results_msg, &executor, &pendulum_motor, &pendulum_controller]() {
+      results_msg->command = *pendulum_controller->get_next_command_message().get();
+      results_msg->state = *pendulum_motor->get_next_sensor_message().get();
+      executor->set_rtt_results_message(results_msg);
+      logger_pub->publish(results_msg);
+    };
+
   // Add a timer to enable regular publication of sensor messages.
   auto motor_publisher_timer = motor_node->create_wall_timer
       (pendulum_motor->get_publish_period(), motor_publish_callback);
   // Add a timer to enable regular publication of command messages.
   auto controller_publisher_timer = controller_node->create_wall_timer
       (pendulum_controller->get_publish_period(), controller_publish_callback);
+  // Add a timer to enable regular publication of results messages.
+  auto logger_publisher_timer = controller_node->create_wall_timer
+      (logger_publisher_period, logger_publish_callback);
```

Comment:
Usually we keep the open parenthesis at the end of the previous line to make the first line more readable. (The two statements above use the same wrapping.)